lib/Driver/Compilation.cpp

```diff
@@ -456,6 +456,69 @@ int Compilation::performJobsImpl() {
         llvm::errs() << Output;
     }
 
+    // In order to handle both old dependencies that have disappeared and new
+    // dependencies that have arisen, we need to reload the dependency file.
+    // Do this whether or not the build succeeded.
+    SmallVector<const Job *, 16> Dependents;
+    if (getIncrementalBuildEnabled()) {
+      const CommandOutput &Output = FinishedCmd->getOutput();
+      StringRef DependenciesFile =
+        Output.getAdditionalOutputForType(types::TY_SwiftDeps);
+      if (!DependenciesFile.empty() &&
+          (ReturnCode == EXIT_SUCCESS || ReturnCode == EXIT_FAILURE)) {
```

Comment:
I interpret [this reference](http://en.cppreference.com/w/cpp/utility/program/EXIT_status) to mean that `ReturnCode` must always be either `EXIT_SUCCESS` or `EXIT_FAILURE`. Is there a case in which it is neither? In other words, is this check necessary?